internal/server/server.go

```diff
@@ -40,13 +40,8 @@ type SOMASServer struct {
 
 // NewSOMASServer returns an instance of the main server we use.
 func NewSOMASServer(gameConfig config.Config) (Server, error) {
-	registeredClients := make(map[shared.ClientID]baseclient.Client, len(baseclient.RegisteredClientFactories))
-	for id, factory := range baseclient.RegisteredClientFactories {
-		registeredClients[id] = factory()
-	}
-
 	clientInfos, clientMap := getClientInfosAndMapFromRegisteredClients(
-		registeredClients,
+		DefaultClientMap(),
```

Comment:
you want a `Factory` method for each team. so that the client pointers containing the state are not preserved between consecutive runs (i.e. each new run has a new clean state)

Reply:
So instead you should write `DefaultClientFactory`

Reply:
Resolved on discord